brave/brave-core
# Search result ad view/click confirmation.

Spec: https://docs.google.com/document/d/1ncuyw0Yv_qawR-FM0_e1r4pNU2pGDhS49KfwfHqqrqs/edit#heading=h.hnm67cu7xxpw

See logic description here: https://github.com/brave/security/issues/1128


Resolves https://github.com/brave/brave-browser/issues/23002

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Test case 1
- Turn off Brave Private Ads.
- Do several search requests on search.brave.com until an ad is shown in a SERP.
- Make sure that a search result ad viewed event is not triggered for Brave Ads.
- Click on the ad link.
- Make sure that search result ad clicked event is not triggered for Brave Ads.

### Test case 2
- Turn on Brave Private Ads.
- Do several search requests on search.brave.com until an ad is shown in a SERP.
- Make sure that a search result ad viewed event is triggered. There should be `Viewed search result ad with placement id` message in rewards-internals log.
- Click on the ad link.
- Make sure that a search result ad clicked event is triggered. There should be `Clicked search result ad with placement id` message in rewards-internals log.
